ddtrace/contrib/pymemcache/patch.py

```diff
@@ -9,27 +10,36 @@
 
 
 _Client = pymemcache.client.base.Client
+_hash_Client = pymemcache.client.hash.Client
+
+_HashClient_client_class = None
+if hasattr(pymemcache.client.hash.HashClient, "client_class"):
+    _HashClient_client_class = pymemcache.client.hash.HashClient.client_class
 
 
 def patch():
-    if getattr(pymemcache.client, '_datadog_patch', False):
+    if getattr(pymemcache.client, "_datadog_patch", False):
         return
 
-    setattr(pymemcache.client, '_datadog_patch', True)
-    setattr(pymemcache.client.base, 'Client', WrappedClient)
+    setattr(pymemcache.client, "_datadog_patch", True)
+    setattr(pymemcache.client.base, "Client", WrappedClient)
+    setattr(pymemcache.client.hash, "Client", WrappedClient)
+    if _HashClient_client_class:
+        pymemcache.client.hash.HashClient.client_class = WrappedClient
```

Comment:
L26-28 are the main changes for supporting `HashClient` and newer versions of the library.

The problem we have is `pymemcache.client.hash` does `from pymemcache.client.base import Client`, so our current patching method of `setattr(pymemcache.client.base, "Client", WrappedClient)` doesn't work since the original `Client` has already been imported into the scope of `pymemcache.client.hash`.

As well, newer versions of the library also have `HashClient.default_class = Client` and we need to be sure it is updated there as well.


We really need to rewrite/update this integration to be more inline with how our other integrations work, but I wanted to make the minimum changes now to support `HashClient` and newer versions of the library

Reply:
> As well, newer versions of the library also have `HashClient.default_class = Client` and we need to be sure it is updated there as well.

Do you mean newer versions of `pymemcache` or our library? I'm curious where we need to update exactly

Reply:
Newer versions of `pymemcache`.

We mostly need to do this "update all places where `Client` exists" because our `pymemcache` integration replaces the entire class with a wrapped class, rather than replacing the methods with wrapped versions.

If we only replaced the necessary methods then all references to the original `Client` class would get updated, but instead we need to go update all places where the `Client` class is imported/used.

This is something that we need to improve with a rewrite, but again this change was aiming more for minimal changes to get `HashClient` and newer versions of the library working.